cudaaligner/src/hirschberg_myers_gpu.cu

```diff
@@ -312,44 +314,51 @@ myers_compute_scores(
         {
             if (threadIdx.x == 0)
                 score(0, reverse ? 1 : 0) = query_size;
-            __syncwarp();
         }
     }
 
+    const int32_t n_warp_iterations = ceiling_divide(n_words, warp_size) * warp_size;
+    __syncwarp();
     for (int32_t t = 1; t <= target_size; ++t)
     {
         int32_t warp_carry = 0;
         if (threadIdx.x == 0)
             warp_carry = 1; // for global alignment the (implicit) first row has to be 0,1,2,3,... -> carry 1
-        for (int32_t idx = threadIdx.x; idx < n_words; idx += warp_size)
+        for (int32_t idx = threadIdx.x; idx < n_warp_iterations; idx += warp_size)
         {
-            const uint32_t warp_mask = idx / warp_size < n_words / warp_size ? 0xffff'ffffu : (1u << (n_words % warp_size)) - 1;
+            if (idx < n_words)
+            {
+                const uint32_t warp_mask = idx / warp_size < n_words / warp_size ? 0xffff'ffffu : (1u << (n_words % warp_size)) - 1;
 
-            WordType pv_local = pv(idx, full_score_matrix ? t - 1 : 0);
-            WordType mv_local = mv(idx, full_score_matrix ? t - 1 : 0);
-            // TODO these might be cached or only computed for the specific t at hand.
-            const WordType highest_bit = WordType(1) << (idx == (n_words - 1) ? (query_end - query_begin) - (n_words - 1) * word_size - 1 : word_size - 1);
+                WordType pv_local = pv(idx, full_score_matrix ? t - 1 : 0);
+                WordType mv_local = mv(idx, full_score_matrix ? t - 1 : 0);
+                // TODO these might be cached or only computed for the specific t at hand.
+                const WordType highest_bit = WordType(1) << (idx == (n_words - 1) ? (query_end - query_begin) - (n_words - 1) * word_size - 1 : word_size - 1);
 
-            const WordType eq = get_query_pattern(query_patterns, idx, pattern_idx_offset, target_begin[reverse ? target_size - t : t - 1], reverse);
+                const WordType eq = get_query_pattern(query_patterns, idx, pattern_idx_offset, target_begin[reverse ? target_size - t : t - 1], reverse);
 
-            warp_carry = myers_advance_block(warp_mask, highest_bit, eq, pv_local, mv_local, warp_carry);
-            if (full_score_matrix)
-            {
-                score(idx, t) = score(idx, t - 1) + warp_carry;
-            }
-            else
-            {
-                if (idx + 1 == n_words)
-                    score(t, reverse ? 1 : 0) = score(t - 1, reverse ? 1 : 0) + warp_carry;
+                warp_carry = myers_advance_block(warp_mask, highest_bit, eq, pv_local, mv_local, warp_carry);
+                if (full_score_matrix)
+                {
+                    score(idx, t) = score(idx, t - 1) + warp_carry;
+                }
+                else
+                {
+                    if (idx + 1 == n_words)
```

Comment:
Pedantic comment - I'm a bit surprised our style check doesn't change this to 
```
if (...)
{
...
}
```

safer to change this to have explicit scoping with `{}`?

Reply:
there are more instances of this on line 350 and 356

Reply:
Old habits die hard. fixed.